serving/docs/adapters.md

```diff
@@ -0,0 +1,180 @@
+
+1. If you are using workflow adapters, there is no need to specify an adapter as it is included in the workflow.
+   Instead, just call the workflow as normal.
+2. Each technique must be implemented by the model handler by parsing the adapter from the Input parameter, content, or body respectively.
+   Our built-in implementations support it from all options
+3. Some of these techniques may not work in all situations. For example, only the custom attributes strategy will work in Amazon SageMaker as it blocks the other options.
+
+
+### (Recommended) Adapters parameter Calling
+
+This passes the adapter as part of the requeset body.
+It will also work with client-side batching and will allow multiple adapters to be passed with one for each input.
+
+```
+
+curl -X POST http://127.0.0.1:8080/invocations \
+    -H "Content-Type: application/json" \
+    -H "X-Amzn-SageMaker-Target-Model: base-1.tar.gz" \
+    -d '{"inputs": ["How is the weather"], "adapters": ["adapter_1"], "parameters": {"max_new_tokens": 25}}'
+```
+
+### Input Content (and query parameter)
+
+This passes the adapter through a query parameter.
+It is reflected in the Input content.
+This will not work in Amazon SageMaker.
+
+```
+curl -X POST http://127.0.0.1:8080/invocations?adapter=adapter_1 \
+    -H "Content-Type: application/json" \
+    -H "X-Amzn-SageMaker-Target-Model: base-1.tar.gz" \
+    -d '{"inputs": ["How is the weather"], "parameters": {"max_new_tokens": 25}}'
+```
+
+### SageMaker Custom Attributes
+
+This passes the adapter through the use of the SageMaker Custom Attributes header.
+It is reflected in the Input properties.
+This option will work in Amazon SageMaker.
+
+```
+curl -X POST http://127.0.0.1:8080/invocations \
+    -H "Content-Type: application/json" \
+    -H "X-Amzn-SageMaker-Target-Model: base-1.tar.gz" \
+    -H "X-Amzn-SageMaker-Custom-Attributes: adapter=adapter_1"
+    -d '{"inputs": ["How is the weather"], "parameters": {"max_new_tokens": 25}}'
+```
+
+## Models with Adapters
+
+There are two kinds of LoRA that can be put onto various engines.
+
+* **Merged LoRA** - This will apply the adapter by modifying the base model in place.
+  It has zero added latency during execution, but has a cost to apply or unapply the merge.
+  It works best for cases with only a few adapters.
+  It is best for single adapter batches, but doesn’t support multi-adapter batches
+* **Unmerged LoRA** - This will alter the model operators to factor in the adapters without changing the base model.
+  It has a higher inference latency for the additional adapter operations.
+  However, it does support multi-adapter batches.
+  It works best for use cases with large numbers of adapters.
+
+With our default handlers, we currently support unmerged LoRA for CausalLM through the huggingface handler.
+Support for other model types and handlers is coming soon.
+
+### Writing Adapter Models
+
+Right now, adapters are only supported through our Python engine.
```

Comment:
we can narrow it down to HF Accelerate. Python engine also includes other backends like Triton core.

Reply:
This is referring to the non-Python engine pieces such as the DJL OnnxRuntime engine or DJL MXNet engine. With the python engine, they could support it in other engines if they were willing to write their own handler instead of using the default ones